### PR TITLE
GEOMESA-2252,GEOMESA-2253 Docs - converter and NiFi updates

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Deprecated 'validation-mode' in favor of 'error-mode' to improve clarity
* Updated spark version to 2.2.x and parameterized it in common.py

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: docs/tutorials/geomesa-quickstart-nifi.rst [hunk 1]
```diff
@@ -27,36 +27,26 @@ Before you begin, you must have the following:
...
-
-The `GDELT Event database <http://www.gdeltproject.org>`__ provides a
-comprehensive time- and location-indexed archive of events reported in
-broadcast, print, and web news media worldwide from 1979 to today. You
-can download raw GDELT data files at
-http://data.gdeltproject.org/events/index.html. For example, the
-following Bash commands use ``wget`` to download all event files for
-2013 (2 GB compressed, 14 GB uncompressed) and ``md5sum`` to verify each
-by their MD5 hash:
+In this tutorial we will be ingesting GDELT data. If you already have some GDELT data available, then
```
Comment (jnh5y) [inline docs/tutorials/geomesa-quickstart-nifi.rst]: By 'available', do you mean ingested?

File: docs/user/convert/jdbc.rst [hunk 2]
```diff
@@ -1,28 +1,52 @@
...
-GeoMesa includes a converter designed to read SQL data using standard JDBC libraries. The connection string
-is specified using the ``connection`` element.
+The JDBC converter allows you to create SimpleFeatures directly from a SQL select statement against an
+eixsting database, using standard JDBC libraries. To use the JDBC converter, specify ``type = "jdbc"``
+in your converter definition.
 
+Configuration
+-------------
+
+The JDBC connection string used to connect to the database must be specified using the ``connection`` element.
```
Comment (jnh5y) [inline docs/user/convert/jdbc.rst]: Can we give a more complete example here?

File: docs/user/spark/jupyter.rst [hunk 3]
```diff
@@ -47,7 +52,8 @@ or
 Installing the Toree Kernel
 ---------------------------
 
-Toree version 0.2.0-dev1 supports Spark 2.0 and above, and should be built from source as described via its `README file`_. Check out the project from GitHub:
+Toree version 0.2.0-dev1 supports Spark 2.0 and above, and should be built from source as described via
```
Comment (jnh5y) [inline docs/user/spark/jupyter.rst]: @aheyne @tkunicki We might should review this bit about using Toree...

File: docs/tutorials/geomesa-quickstart-nifi.rst (hunk 1 above)
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-nifi.rst]: downloaded, i'll use that term instead

File: docs/user/convert/jdbc.rst (hunk 2 above)
Comment (elahrvivaz) [reply docs/user/convert/jdbc.rst]: added

File: docs/user/spark/jupyter.rst (hunk 3 above)
Comment (aheyne) [reply docs/user/spark/jupyter.rst]: IMO we should just ditch Toree and/or strongly recommend Zeppelin for anything Scala and Jupyter for only Python uses.

**Pull request merged**